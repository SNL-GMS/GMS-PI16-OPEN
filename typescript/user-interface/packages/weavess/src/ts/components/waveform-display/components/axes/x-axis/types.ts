export interface XAxisProps {
  /** Epoch seconds end */
  endTimeSecs: number;

  /** Epoch seconds start */
  startTimeSecs: number;

  /** Add border to top */
  borderTop: boolean;

  /** Label width in px */
  labelWidthPx: number;

  /** Scrollbar width in px */
  scrollbarWidthPx: number;

  /** (optional) x-axis label  */
  label?: string;

  /** Call back to get the current view range */
  getViewRange(): [number, number];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface XAxisState {}
