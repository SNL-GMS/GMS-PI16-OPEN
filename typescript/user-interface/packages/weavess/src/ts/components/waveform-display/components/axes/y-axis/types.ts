export interface YAxisProps {
  /** The height in percentage */
  heightInPercentage: number;

  /** Max amplitude as a number */
  maxAmplitude: number;

  /** Min amplitude as a number */
  minAmplitude: number;

  /** the y axis ticks */
  yAxisTicks?: number[];
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface YAxisState {}
