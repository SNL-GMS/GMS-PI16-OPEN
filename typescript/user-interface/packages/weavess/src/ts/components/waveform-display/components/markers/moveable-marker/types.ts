import { WeavessTypes } from '@gms/weavess-core';

export interface MoveableMarkerProps {
  /** The marker config */
  marker: WeavessTypes.Marker;

  /** The associated start marker (usually only for a selection window) */
  associatedStartMarker?: WeavessTypes.Marker;

  /** The associated end marker (usually only for a selection window) */
  associatedEndMarker?: WeavessTypes.Marker;

  /** Percentage Location 0-100 as number */
  percentageLocation: number;

  /** Label Width in px */
  labelWidthPx: number;

  /** react key but called name since key was causing fortify issues */
  name?: string;

  /** Start and end of the entire time range */
  timeRange(): WeavessTypes.TimeRange;

  /** current view port of the time range */
  viewTimeRange(): WeavessTypes.TimeRange;

  /** Returns container client width */
  containerClientWidth(): number;

  /** Returns the viewPort client width */
  viewportClientWidth(): number;

  /**
   * (optional) updates the location of the marker
   *
   * @param marker the marker
   */
  onUpdateMarker?(marker: WeavessTypes.Marker): void;

  /**  */
  updateTimeWindowSelection?(): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface MoveableMarkerState {}
