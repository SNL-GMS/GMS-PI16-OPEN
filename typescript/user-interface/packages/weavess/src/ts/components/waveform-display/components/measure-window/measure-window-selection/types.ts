import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';
import * as React from 'react';

export interface MeasureWindowSelectionListenerProps {
  displayTimeRange: WeavessTypes.TimeRange;
  hotKeys: WeavessTypes.HotKeysConfiguration;
  isMeasureWindowEnabled(): boolean;
  children(measureWindowParams: {
    contentRenderer: JSX.Element;
    onMouseDown: (e: React.MouseEvent<HTMLDivElement>) => void;
  }): JSX.Element | null;
  toast(
    message: string,
    intent?: Intent | undefined,
    icon?: IconName | undefined,
    timeout?: number | undefined
  ): void;
  updateMeasureWindowPanel(
    timeRange: WeavessTypes.TimeRange,
    removeMeasureWindowSelection: () => void
  ): void;
  computeTimeSecsFromMouseXPixels(mouseXPx: number): number;
}
