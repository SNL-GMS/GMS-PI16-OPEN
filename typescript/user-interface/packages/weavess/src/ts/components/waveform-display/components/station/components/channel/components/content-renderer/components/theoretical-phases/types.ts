import { IconName, Intent } from '@blueprintjs/core';
import { WeavessTypes } from '@gms/weavess-core';

export interface TheoreticalPhasesProps {
  /** Station Id as string */
  stationId: string;

  /** The theoretical phase windows */
  theoreticalPhaseWindows: WeavessTypes.TheoreticalPhaseWindow[] | undefined;

  /** Boolean is default channel */
  isDefaultChannel: boolean;

  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Epoch seconds end time */
  displayEndTimeSecs: number;

  /** (optional) callback events Ex on waveform click */
  events?: WeavessTypes.ChannelContentEvents;

  /**
   * Call back for toast, which is the notification pop up
   *
   * @param message message to be displayed as a string
   * @param intent (optional) NONE, PRIMARY, WARNING, DANGER ex NONE = "none"
   * @param icon (optional) the way it looks
   * @param timeout (optional) time before message disappears
   */
  toast(message: string, intent?: Intent, icon?: IconName, timeout?: number): void;

  /**
   * Returns the time in seconds for the given clientX.
   *
   * @param clientX The clientX
   *
   * @returns The time in seconds; undefined if clientX is
   * out of the channel's bounds on screen.
   */
  getTimeSecsForClientX(clientX: number): number | undefined;

  /**
   * Toggle display of the drag indicator for this channel
   *
   * @param show True to show drag indicator
   * @param color The color of the drag indicator
   */
  toggleDragIndicator(show: boolean, color: string): void;

  /**
   * Set the position for the drag indicator
   *
   * @param clientX The clientX
   */
  positionDragIndicator(clientX: number): void;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TheoreticalPhasesState {}
