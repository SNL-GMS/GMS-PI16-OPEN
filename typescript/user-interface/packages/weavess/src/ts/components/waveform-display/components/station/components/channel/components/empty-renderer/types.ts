export interface EmptyRendererProps {
  /** Epoch seconds start time */
  displayStartTimeSecs: number;

  /** Boolean is default channel */
  displayEndTimeSecs: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyRendererState {}
