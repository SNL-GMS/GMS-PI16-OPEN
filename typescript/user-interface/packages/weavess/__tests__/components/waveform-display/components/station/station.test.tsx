/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import { DistanceUnits } from '@gms/weavess-core/lib/types';
import * as Enzyme from 'enzyme';
import * as React from 'react';

import { Station } from '../../../../../src/ts/components/waveform-display/components/station/station';
import {
  CommonStationProps,
  StationProps
} from '../../../../../src/ts/components/waveform-display/components/station/types';
import { defaultConfiguration } from '../../../../../src/ts/components/waveform-display/configuration';

const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);

const commonProps: CommonStationProps = {
  converters: {
    computeFractionOfCanvasFromMouseXPx: jest.fn(),
    computeTimeSecsForMouseXFractionalPosition: jest.fn(),
    computeTimeSecsFromMouseXPixels: jest.fn()
  },
  getCurrentViewRangeInSeconds: jest.fn(),
  onMouseDown: jest.fn(),
  onMouseMove: jest.fn(),
  onMouseUp: jest.fn(),
  renderWaveforms: jest.fn(),
  onContextMenu: jest.fn(),
  updateMeasureWindow: jest.fn(),
  isMeasureWindow: false
};

const props: StationProps = {
  ...commonProps,
  initialConfiguration: defaultConfiguration,
  station: {
    id: 'station',
    defaultChannel: {
      id: 'default-channel',
      name: 'default-channel'
    },
    name: 'station',
    distance: 0,
    distanceUnits: DistanceUnits.km,
    nonDefaultChannels: [
      {
        id: 'channel',
        name: 'channel',
        waveform: {
          channelSegmentId: '1',
          channelSegments: new Map(),
          masks: [
            {
              id: 'mask',
              color: 'ff0000',
              startTimeSecs: 5,
              endTimeSecs: 10
            }
          ]
        }
      }
    ]
  },
  displayStartTimeSecs: 0,
  displayEndTimeSecs: 500,
  shouldRenderWaveforms: false,
  shouldRenderSpectrograms: false,
  workerRpcs: [],
  selections: {
    channels: [],
    predictedPhases: [],
    signalDetections: []
  },
  getCanvasBoundingRect: jest.fn(() => new DOMRect(0, 0, 100, 100)),
  updateVisibleChannelForStation: jest.fn(),
  events: undefined,
  glMin: 0,
  glMax: 100,
  canvasRef: jest.fn(),
  getViewRange: jest.fn(),
  toast: jest.fn()
};

const wrapper = Enzyme.mount(<Station {...props} />);
const instance: Station = wrapper.find(Station).instance() as Station;

describe('Weavess Station', () => {
  it('to be defined', () => {
    expect(Station).toBeDefined();
  });

  it('shallow renders', () => {
    const shallow = Enzyme.shallow(<Station {...props} />);
    expect(shallow).toMatchSnapshot();

    expect(Enzyme.shallow(<Station {...props} />)).toMatchSnapshot();
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidCatch', () => {
    const spy = jest.spyOn(instance, 'componentDidCatch');
    instance.componentDidCatch(new Error('error'), { componentStack: undefined });
    expect(spy).toBeCalled();
  });

  it('reset amplitude', () => {
    const spy = jest.spyOn(instance, 'resetAmplitude');
    instance.resetAmplitude();
    expect(spy).toBeCalled();
  });

  it('update mask labels', () => {
    const station = new Station(props);
    const spy = jest.spyOn(station, 'updateMaskLabels');
    station.updateMaskLabels();
    expect(spy).toBeCalled();
  });
});
