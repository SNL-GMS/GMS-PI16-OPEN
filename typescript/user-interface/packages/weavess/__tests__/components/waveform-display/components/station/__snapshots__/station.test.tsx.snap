// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess Station renders 1`] = `
<Station
  canvasRef={
    [MockFunction] {
      "calls": Array [
        Array [],
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  converters={
    Object {
      "computeFractionOfCanvasFromMouseXPx": [MockFunction],
      "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
      "computeTimeSecsFromMouseXPixels": [MockFunction],
    }
  }
  displayEndTimeSecs={500}
  displayStartTimeSecs={0}
  getCanvasBoundingRect={[MockFunction]}
  getCurrentViewRangeInSeconds={[MockFunction]}
  getViewRange={[MockFunction]}
  glMax={100}
  glMin={0}
  initialConfiguration={
    Object {
      "colorScale": undefined,
      "defaultChannel": Object {
        "disableMaskModification": false,
        "disableMeasureWindow": false,
        "disablePreditedPhaseModification": false,
        "disableSignalDetectionModification": false,
      },
      "defaultChannelHeightPx": 50,
      "hotKeys": Object {
        "amplitudeScale": "KeyS",
        "amplitudeScaleReset": "Alt+Shift+KeyS",
        "amplitudeScaleSingleReset": "Alt+KeyS",
        "drawMeasureWindow": "Alt",
        "maskCreate": "KeyM",
      },
      "labelWidthPx": 184,
      "nonDefaultChannel": Object {
        "disableMaskModification": false,
        "disableMeasureWindow": false,
        "disablePreditedPhaseModification": false,
        "disableSignalDetectionModification": false,
      },
      "shouldRenderSpectrograms": true,
      "shouldRenderWaveforms": true,
      "xAxisLabel": undefined,
    }
  }
  isMeasureWindow={false}
  onContextMenu={[MockFunction]}
  onMouseDown={[MockFunction]}
  onMouseMove={[MockFunction]}
  onMouseUp={[MockFunction]}
  renderWaveforms={[MockFunction]}
  selections={
    Object {
      "channels": Array [],
      "predictedPhases": Array [],
      "signalDetections": Array [],
    }
  }
  shouldRenderSpectrograms={false}
  shouldRenderWaveforms={false}
  station={
    Object {
      "defaultChannel": Object {
        "id": "default-channel",
        "name": "default-channel",
      },
      "distance": 0,
      "distanceUnits": "km",
      "id": "station",
      "name": "station",
      "nonDefaultChannels": Array [
        Object {
          "id": "channel",
          "name": "channel",
          "waveform": Object {
            "channelSegmentId": "1",
            "channelSegments": Map {},
            "masks": Array [
              Object {
                "color": "ff0000",
                "endTimeSecs": 10,
                "id": "mask",
                "startTimeSecs": 5,
              },
            ],
          },
        },
      ],
    }
  }
  toast={[MockFunction]}
  updateMeasureWindow={[MockFunction]}
  updateVisibleChannelForStation={[MockFunction]}
  workerRpcs={Array []}
>
  <div
    className="station"
    style={
      Object {
        "height": 51,
      }
    }
  >
    <Channel
      canvasRef={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      channel={
        Object {
          "id": "default-channel",
          "name": "default-channel",
        }
      }
      converters={
        Object {
          "computeFractionOfCanvasFromMouseXPx": [MockFunction],
          "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
          "computeTimeSecsFromMouseXPixels": [MockFunction],
        }
      }
      displayEndTimeSecs={500}
      displayStartTimeSecs={0}
      distance={0}
      distanceUnits="km"
      expanded={false}
      getCanvasBoundingRect={[MockFunction]}
      getCurrentViewRangeInSeconds={[MockFunction]}
      getViewRange={[MockFunction]}
      glMax={100}
      glMin={0}
      height={50}
      index={0}
      initialConfiguration={
        Object {
          "colorScale": undefined,
          "defaultChannel": Object {
            "disableMaskModification": false,
            "disableMeasureWindow": false,
            "disablePreditedPhaseModification": false,
            "disableSignalDetectionModification": false,
          },
          "defaultChannelHeightPx": 50,
          "hotKeys": Object {
            "amplitudeScale": "KeyS",
            "amplitudeScaleReset": "Alt+Shift+KeyS",
            "amplitudeScaleSingleReset": "Alt+KeyS",
            "drawMeasureWindow": "Alt",
            "maskCreate": "KeyM",
          },
          "labelWidthPx": 184,
          "nonDefaultChannel": Object {
            "disableMaskModification": false,
            "disableMeasureWindow": false,
            "disablePreditedPhaseModification": false,
            "disableSignalDetectionModification": false,
          },
          "shouldRenderSpectrograms": true,
          "shouldRenderWaveforms": true,
          "xAxisLabel": undefined,
        }
      }
      isDefaultChannel={true}
      isExpandable={true}
      isMeasureWindow={false}
      key="station-default-channel-default-channel"
      offsetSecs={0}
      onContextMenu={[MockFunction]}
      onMouseDown={[MockFunction]}
      onMouseMove={[Function]}
      onMouseUp={[MockFunction]}
      renderWaveforms={[MockFunction]}
      selections={
        Object {
          "channels": Array [],
          "predictedPhases": Array [],
          "signalDetections": Array [],
        }
      }
      shouldRenderSpectrograms={false}
      shouldRenderWaveforms={false}
      showMaskIndicator={true}
      stationId="station"
      toast={[MockFunction]}
      toggleExpansion={[Function]}
      updateMeasureWindow={[MockFunction]}
      workerRpcs={Array []}
    >
      <div
        className="channel"
        data-cy="default-channel-channel"
        onKeyDown={[Function]}
        style={
          Object {
            "height": "50px",
            "maxHeight": "50px",
          }
        }
        tabIndex={0}
      >
        <div
          className="channel-label-container"
          data-cy="default-channel-label"
          style={
            Object {
              "height": "50px",
              "width": "184px",
            }
          }
          tabIndex={0}
        >
          <Label
            canvasRef={
              [MockFunction] {
                "calls": Array [
                  Array [],
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            channel={
              Object {
                "id": "default-channel",
                "name": "default-channel",
              }
            }
            channelName="default-channel"
            converters={
              Object {
                "computeFractionOfCanvasFromMouseXPx": [MockFunction],
                "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
                "computeTimeSecsFromMouseXPixels": [MockFunction],
              }
            }
            displayEndTimeSecs={500}
            displayStartTimeSecs={0}
            distance={0}
            distanceUnits="km"
            expanded={false}
            getCanvasBoundingRect={[MockFunction]}
            getCurrentViewRangeInSeconds={[MockFunction]}
            getViewRange={[MockFunction]}
            glMax={100}
            glMin={0}
            height={50}
            index={0}
            initialConfiguration={
              Object {
                "colorScale": undefined,
                "defaultChannel": Object {
                  "disableMaskModification": false,
                  "disableMeasureWindow": false,
                  "disablePreditedPhaseModification": false,
                  "disableSignalDetectionModification": false,
                },
                "defaultChannelHeightPx": 50,
                "hotKeys": Object {
                  "amplitudeScale": "KeyS",
                  "amplitudeScaleReset": "Alt+Shift+KeyS",
                  "amplitudeScaleSingleReset": "Alt+KeyS",
                  "drawMeasureWindow": "Alt",
                  "maskCreate": "KeyM",
                },
                "labelWidthPx": 184,
                "nonDefaultChannel": Object {
                  "disableMaskModification": false,
                  "disableMeasureWindow": false,
                  "disablePreditedPhaseModification": false,
                  "disableSignalDetectionModification": false,
                },
                "shouldRenderSpectrograms": true,
                "shouldRenderWaveforms": true,
                "xAxisLabel": undefined,
              }
            }
            isDefaultChannel={true}
            isExpandable={true}
            isMeasureWindow={false}
            offsetSecs={0}
            onContextMenu={[MockFunction]}
            onMouseDown={[MockFunction]}
            onMouseMove={[Function]}
            onMouseUp={[MockFunction]}
            renderWaveforms={[MockFunction]}
            selections={
              Object {
                "channels": Array [],
                "predictedPhases": Array [],
                "signalDetections": Array [],
              }
            }
            shouldRenderSpectrograms={false}
            shouldRenderWaveforms={false}
            showMaskIndicator={true}
            stationId="station"
            toast={[MockFunction]}
            toggleExpansion={[Function]}
            updateMeasureWindow={[MockFunction]}
            workerRpcs={Array []}
            yAxisBounds={Array []}
          >
            <div
              className="label"
              onContextMenu={[Function]}
              tabIndex={0}
            >
              <div
                className="label-container"
              >
                <div
                  className="label-container-left-parent"
                >
                  <div
                    className="label-container-left-parent-expansion-button"
                    data-cy="weavess-expand-parent"
                    data-cy-channel-name="default-channel"
                    onClick={[Function]}
                  >
                    +
                  </div>
                </div>
                <div
                  className="label-container-content"
                >
                  <div
                    className="label-container-content-label"
                    data-cy="channel-label-default-channel"
                    onClick={[Function]}
                  >
                    <span>
                      default-channel
                      <div>
                        <p>
                           
                          <span
                            className="label-container-content-mask-indicator"
                            key=""
                          >
                            M
                          </span>
                        </p>
                      </div>
                    </span>
                  </div>
                  <div
                    data-cy-contains-amplitude-markers={false}
                    style={
                      Object {
                        "height": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </Label>
        </div>
        <div
          className="channel-content-container"
          style={
            Object {
              "height": "50px",
              "left": "184px",
              "width": "calc(100% - 184px)",
            }
          }
        >
          <ContentRenderer
            canvasRef={
              [MockFunction] {
                "calls": Array [
                  Array [],
                  Array [],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            channel={
              Object {
                "id": "default-channel",
                "name": "default-channel",
              }
            }
            channelId="default-channel"
            color="#4580E6"
            contentRenderers={Array []}
            converters={
              Object {
                "computeFractionOfCanvasFromMouseXPx": [MockFunction],
                "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
                "computeTimeSecsFromMouseXPixels": [MockFunction],
              }
            }
            disablePredictedPhaseModification={false}
            disableSignalDetectionModification={false}
            displayEndTimeSecs={500}
            displayStartTimeSecs={0}
            displayType={
              Array [
                "LINE",
              ]
            }
            distance={0}
            distanceUnits="km"
            expanded={false}
            getCanvasBoundingRect={[MockFunction]}
            getCurrentViewRangeInSeconds={[MockFunction]}
            getViewRange={[MockFunction]}
            glMax={100}
            glMin={0}
            height={50}
            index={0}
            initialConfiguration={
              Object {
                "colorScale": undefined,
                "defaultChannel": Object {
                  "disableMaskModification": false,
                  "disableMeasureWindow": false,
                  "disablePreditedPhaseModification": false,
                  "disableSignalDetectionModification": false,
                },
                "defaultChannelHeightPx": 50,
                "hotKeys": Object {
                  "amplitudeScale": "KeyS",
                  "amplitudeScaleReset": "Alt+Shift+KeyS",
                  "amplitudeScaleSingleReset": "Alt+KeyS",
                  "drawMeasureWindow": "Alt",
                  "maskCreate": "KeyM",
                },
                "labelWidthPx": 184,
                "nonDefaultChannel": Object {
                  "disableMaskModification": false,
                  "disableMeasureWindow": false,
                  "disablePreditedPhaseModification": false,
                  "disableSignalDetectionModification": false,
                },
                "shouldRenderSpectrograms": true,
                "shouldRenderWaveforms": true,
                "xAxisLabel": undefined,
              }
            }
            isDefaultChannel={true}
            isExpandable={true}
            isMeasureWindow={false}
            offsetSecs={0}
            onContextMenu={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            pointSize={2}
            renderWaveforms={[MockFunction]}
            selections={
              Object {
                "channels": Array [],
                "predictedPhases": Array [],
                "signalDetections": Array [],
              }
            }
            setYAxisBounds={[Function]}
            shouldRenderSpectrograms={false}
            shouldRenderWaveforms={false}
            showMaskIndicator={true}
            stationId="station"
            toast={[MockFunction]}
            toggleExpansion={[Function]}
            updateMeasureWindow={[MockFunction]}
            workerRpcs={Array []}
          >
            <div
              className="contentrenderer"
              onContextMenu={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "backgroundColor": "initial",
                }
              }
              tabIndex={0}
            >
              <EmptyRenderer
                canvasRef={
                  [MockFunction] {
                    "calls": Array [
                      Array [],
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  }
                }
                channel={
                  Object {
                    "id": "default-channel",
                    "name": "default-channel",
                  }
                }
                converters={
                  Object {
                    "computeFractionOfCanvasFromMouseXPx": [MockFunction],
                    "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
                    "computeTimeSecsFromMouseXPixels": [MockFunction],
                  }
                }
                displayEndTimeSecs={500}
                displayStartTimeSecs={0}
                distance={0}
                distanceUnits="km"
                expanded={false}
                getCanvasBoundingRect={[MockFunction]}
                getCurrentViewRangeInSeconds={[MockFunction]}
                getViewRange={[MockFunction]}
                glMax={100}
                glMin={0}
                height={50}
                index={0}
                initialConfiguration={
                  Object {
                    "colorScale": undefined,
                    "defaultChannel": Object {
                      "disableMaskModification": false,
                      "disableMeasureWindow": false,
                      "disablePreditedPhaseModification": false,
                      "disableSignalDetectionModification": false,
                    },
                    "defaultChannelHeightPx": 50,
                    "hotKeys": Object {
                      "amplitudeScale": "KeyS",
                      "amplitudeScaleReset": "Alt+Shift+KeyS",
                      "amplitudeScaleSingleReset": "Alt+KeyS",
                      "drawMeasureWindow": "Alt",
                      "maskCreate": "KeyM",
                    },
                    "labelWidthPx": 184,
                    "nonDefaultChannel": Object {
                      "disableMaskModification": false,
                      "disableMeasureWindow": false,
                      "disablePreditedPhaseModification": false,
                      "disableSignalDetectionModification": false,
                    },
                    "shouldRenderSpectrograms": true,
                    "shouldRenderWaveforms": true,
                    "xAxisLabel": undefined,
                  }
                }
                isDefaultChannel={true}
                isExpandable={true}
                isMeasureWindow={false}
                offsetSecs={0}
                onContextMenu={[MockFunction]}
                onMouseDown={[MockFunction]}
                onMouseMove={[Function]}
                onMouseUp={[MockFunction]}
                renderWaveforms={[MockFunction]}
                selections={
                  Object {
                    "channels": Array [],
                    "predictedPhases": Array [],
                    "signalDetections": Array [],
                  }
                }
                shouldRenderSpectrograms={false}
                shouldRenderWaveforms={false}
                showMaskIndicator={true}
                stationId="station"
                toast={[MockFunction]}
                toggleExpansion={[Function]}
                updateMeasureWindow={[MockFunction]}
                workerRpcs={Array []}
              />
              <div
                className="contentrenderer-content contentrenderer-content--sticky"
                style={
                  Object {
                    "width": "calc(100vw - 184)",
                  }
                }
              />
              <div
                className="contentrenderer-content"
                style={
                  Object {
                    "left": 184,
                    "width": "calc(100% - 184)",
                  }
                }
              >
                <div
                  className="contentrenderer-content-drag-indicator"
                />
                <div
                  className="contentrenderer-content-markers"
                />
                <SignalDetections
                  channelId="default-channel"
                  disableModification={false}
                  displayEndTimeSecs={500}
                  displayStartTimeSecs={0}
                  getTimeSecsForClientX={[Function]}
                  isDefaultChannel={true}
                  positionDragIndicator={[Function]}
                  selectedSignalDetections={Array []}
                  stationId="station"
                  toast={[MockFunction]}
                  toggleDragIndicator={[Function]}
                />
                <PredictedPhases
                  channelId="default-channel"
                  disableModification={false}
                  displayEndTimeSecs={500}
                  displayStartTimeSecs={0}
                  getTimeSecsForClientX={[Function]}
                  isDefaultChannel={true}
                  positionDragIndicator={[Function]}
                  selectedPredictedPhases={Array []}
                  stationId="station"
                  toast={[MockFunction]}
                  toggleDragIndicator={[Function]}
                />
                <TheoreticalPhases
                  displayEndTimeSecs={500}
                  displayStartTimeSecs={0}
                  getTimeSecsForClientX={[Function]}
                  isDefaultChannel={true}
                  positionDragIndicator={[Function]}
                  stationId="station"
                  toast={[MockFunction]}
                  toggleDragIndicator={[Function]}
                />
              </div>
            </div>
          </ContentRenderer>
        </div>
      </div>
    </Channel>
  </div>
</Station>
`;

exports[`Weavess Station shallow renders 1`] = `
<div
  className="station"
  style={
    Object {
      "height": 51,
    }
  }
>
  <Channel
    canvasRef={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    channel={
      Object {
        "id": "default-channel",
        "name": "default-channel",
      }
    }
    converters={
      Object {
        "computeFractionOfCanvasFromMouseXPx": [MockFunction],
        "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
        "computeTimeSecsFromMouseXPixels": [MockFunction],
      }
    }
    displayEndTimeSecs={500}
    displayStartTimeSecs={0}
    distance={0}
    distanceUnits="km"
    expanded={false}
    getCanvasBoundingRect={[MockFunction]}
    getCurrentViewRangeInSeconds={[MockFunction]}
    getViewRange={[MockFunction]}
    glMax={100}
    glMin={0}
    height={50}
    index={0}
    initialConfiguration={
      Object {
        "colorScale": undefined,
        "defaultChannel": Object {
          "disableMaskModification": false,
          "disableMeasureWindow": false,
          "disablePreditedPhaseModification": false,
          "disableSignalDetectionModification": false,
        },
        "defaultChannelHeightPx": 50,
        "hotKeys": Object {
          "amplitudeScale": "KeyS",
          "amplitudeScaleReset": "Alt+Shift+KeyS",
          "amplitudeScaleSingleReset": "Alt+KeyS",
          "drawMeasureWindow": "Alt",
          "maskCreate": "KeyM",
        },
        "labelWidthPx": 184,
        "nonDefaultChannel": Object {
          "disableMaskModification": false,
          "disableMeasureWindow": false,
          "disablePreditedPhaseModification": false,
          "disableSignalDetectionModification": false,
        },
        "shouldRenderSpectrograms": true,
        "shouldRenderWaveforms": true,
        "xAxisLabel": undefined,
      }
    }
    isDefaultChannel={true}
    isExpandable={true}
    isMeasureWindow={false}
    key="station-default-channel-default-channel"
    offsetSecs={0}
    onContextMenu={[MockFunction]}
    onMouseDown={[MockFunction]}
    onMouseMove={[Function]}
    onMouseUp={[MockFunction]}
    renderWaveforms={[MockFunction]}
    selections={
      Object {
        "channels": Array [],
        "predictedPhases": Array [],
        "signalDetections": Array [],
      }
    }
    shouldRenderSpectrograms={false}
    shouldRenderWaveforms={false}
    showMaskIndicator={true}
    stationId="station"
    toast={[MockFunction]}
    toggleExpansion={[Function]}
    updateMeasureWindow={[MockFunction]}
    workerRpcs={Array []}
  />
</div>
`;

exports[`Weavess Station shallow renders 2`] = `
<div
  className="station"
  style={
    Object {
      "height": 51,
    }
  }
>
  <Channel
    canvasRef={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    channel={
      Object {
        "id": "default-channel",
        "name": "default-channel",
      }
    }
    converters={
      Object {
        "computeFractionOfCanvasFromMouseXPx": [MockFunction],
        "computeTimeSecsForMouseXFractionalPosition": [MockFunction],
        "computeTimeSecsFromMouseXPixels": [MockFunction],
      }
    }
    displayEndTimeSecs={500}
    displayStartTimeSecs={0}
    distance={0}
    distanceUnits="km"
    expanded={false}
    getCanvasBoundingRect={[MockFunction]}
    getCurrentViewRangeInSeconds={[MockFunction]}
    getViewRange={[MockFunction]}
    glMax={100}
    glMin={0}
    height={50}
    index={0}
    initialConfiguration={
      Object {
        "colorScale": undefined,
        "defaultChannel": Object {
          "disableMaskModification": false,
          "disableMeasureWindow": false,
          "disablePreditedPhaseModification": false,
          "disableSignalDetectionModification": false,
        },
        "defaultChannelHeightPx": 50,
        "hotKeys": Object {
          "amplitudeScale": "KeyS",
          "amplitudeScaleReset": "Alt+Shift+KeyS",
          "amplitudeScaleSingleReset": "Alt+KeyS",
          "drawMeasureWindow": "Alt",
          "maskCreate": "KeyM",
        },
        "labelWidthPx": 184,
        "nonDefaultChannel": Object {
          "disableMaskModification": false,
          "disableMeasureWindow": false,
          "disablePreditedPhaseModification": false,
          "disableSignalDetectionModification": false,
        },
        "shouldRenderSpectrograms": true,
        "shouldRenderWaveforms": true,
        "xAxisLabel": undefined,
      }
    }
    isDefaultChannel={true}
    isExpandable={true}
    isMeasureWindow={false}
    key="station-default-channel-default-channel"
    offsetSecs={0}
    onContextMenu={[MockFunction]}
    onMouseDown={[MockFunction]}
    onMouseMove={[Function]}
    onMouseUp={[MockFunction]}
    renderWaveforms={[MockFunction]}
    selections={
      Object {
        "channels": Array [],
        "predictedPhases": Array [],
        "signalDetections": Array [],
      }
    }
    shouldRenderSpectrograms={false}
    shouldRenderWaveforms={false}
    showMaskIndicator={true}
    stationId="station"
    toast={[MockFunction]}
    toggleExpansion={[Function]}
    updateMeasureWindow={[MockFunction]}
    workerRpcs={Array []}
  />
</div>
`;
